docs/decisions/infra/0010-feature-flags-system-design.md

```diff
@@ -0,0 +1,124 @@
+
+## Requirements
+
+1. The project team can define feature flags, or feature toggles, that enable/disable a set of functionality in an environment, depending on whether the flag is enabled or disabled.
+2. The feature flagging system should support gradual rollouts, the ability to roll out a feature incrementally to a percentage of users.
+3. Separate feature flag configuration from implementation of the feature flags, so that feature flags can be changed frequently through configuration without touching the underlying feature flag infrastructure code.
+
+## Approach
+
+This tech spec explores the use of [AWS CloudWatch Evidently](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Evidently.html), a service that provides functionality for feature flags, gradual rollouts, and conducting split testing (A/B testing) experiments.
+
+## Feature management
+
+One key design question is how features should be managed once defined. How should a team go about enabling and disabling the feature flags and adjusting the percentage of traffic to send to a new feature during a feature launch?
+
+### Option 1. Manage features using app-config module as part of service layer
+
+Define features in [app-config](/infra/app/app-config/), and use that configuration in the [service layer](/infra/app/service/) to create and configure the features in AWS Evidently.
+
+* Everything is defined in code and in one place.
+* Feature and feature configurations are updated automatically as part of service deploys or can be done manually via a terraform apply.
+
+The configuration in the app-config module might look something like the following:
+
+```terraform
+features = {
+  some_disabled_feature = {} // defaults to enabled = false
+
+  some_enabled_feature = {
+    enabled = true
+  }
+
+  partially_rolled_out_feature = {
+    throttle_percentage = 0.2
+  }
+}
+```
+
+### Option 2. Manage features using app-config as part of a separate infrastructure layer
+
+Define features in [app-config](/infra/app/app-config/main.tf). Create the features in the [service layer](/infra/app/service/) but set things like throttle percentages (for gradual rollouts) in a separate infrastructure layer.
+
+* Allows for separation of permissions. For example, individuals can have permission to update feature launch throttle percentages without having permission to create, edit, or delete the features themselves.
+
+### Option 3. Manage features in AWS Console outside of terraform
```

Comment:
I like Option 3 - one caveat is that in the future we could include the option for wiremock (ie. for load testing or local development - having the ability to mock the evidently call is nice). Additionally, it could make sense to allow cookies set via query params to override evidently in lower envs so folks w/out aws access can easily configure an experience they want for testing.

Reply:
Ah disregard, I see you mention this lower down!